setup-homebrew/main.sh

```diff
@@ -44,68 +44,96 @@ HOMEBREW_PREFIX="$(brew --prefix)"
 HOMEBREW_REPOSITORY="$(brew --repo)"
 HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
 HOMEBREW_CASK_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-cask"
+HOMEBREW_OTHER_CASK_REPOSITORIES=(
+    "${HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-cask-drivers"
+    "${HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-cask-fonts"
+    "${HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-cask-versions"
+)
+if [[ "$GITHUB_REPOSITORY" =~ ^.+/homebrew-.+$ ]]; then
+    HOMEBREW_TAP_REPOSITORY="$(brew --repo "$GITHUB_REPOSITORY")"
+fi
 
 # Do in container or on the runner
 if [[ -f "/.dockerenv" ]] || ([[ -f /proc/1/cgroup ]] && grep -qE "actions_job|docker" /proc/1/cgroup); then
     # Fix permissions to give normal user access
-    sudo chown -R "$(whoami)" "$HOME" "$PWD/.."
+    if [[ -n "$(command -v setfacl)" ]]; then
+        setfacl_dirs=()
+        user=$(whoami)
+        for dir in "$HOME" "$RUNNER_WORKSPACE" "$RUNNER_TEMP"
+        do
+            if [[ ! -w "$dir" ]]; then
+                sudo setfacl -Rm "d:u:$user:rwX,u:$user:rwX" "$dir"
+                setfacl_dirs+=("$dir")
+            fi
+        done
+        echo "SETFACL_DIRECTORIES=$(IFS=:; echo "${setfacl_dirs[*]}")" >> $GITHUB_STATE
+        echo "Set up ACL."
+    elif [[ ! -w "$GITHUB_OUTPUT" ]]; then
+        echo "::warning::Missing write permissions to GitHub directories. Install \`acl\` (\`setfacl\`)."
+    fi
```

Comment:
This is pretty involved. Can it get more comments explaining what's going on, why it's needed, etc?

Reply:
This is the exact same thinng as before with  `chown` but using `setfacl` instead, which works better as it can apply to new files (needed for file commads to work like `GITHUB_OUTPUT`).

`setfacl` _might_ not be installed so I added a check to be safe.

The only other thing happening here is to track what we've changed so that we can cleanup and revert back when the job completes.